### PR TITLE
Add transaction fee api, fix archethic-foundation/archethic-node#196

### PR DESCRIPTION
**Pull request by roychowdhuryrohit-dev**
Added transaction fee api for calculating and sending fee of transaction body received
Added new route for transaction fee api
Added testcases for transaction body validation

### CODE REVIEW

**Review by samuel-uniris — CHANGES_REQUESTED** [review]
Thanks for the PR.
Just one comment regarding the inclusion of the UCO price provider.

File: lib/archethic_web/controllers/api/transaction_controller.ex [hunk 1]
```diff
@@ -86,4 +95,31 @@ defmodule ArchEthicWeb.API.TransactionController do
...
+          |> Mining.get_transaction_fee(
+            OracleChain.get_uco_price(DateTime.utc_now())
+            |> Keyword.fetch!(:usd)
+          )
+
+        conn
+        |> put_status(:ok)
+        |> json(%{
+          "fee" => fee / 100_000_000,
+          "uco_price_provider" => @ucoprice_provider
```
[the remark this answers is not in the document]
Comment (roychowdhuryrohit-dev) [reply lib/archethic_web/controllers/api/transaction_controller.ex:115]: For transparency towards end user, we should provide the information that the calculation has been powered by Coingecko data. But then again it's a personal opinion.

[the remark this answers is not in the document]
Comment (samuel-uniris) [reply lib/archethic_web/controllers/api/transaction_controller.ex:115]: We can provide information about the rates of the UCO price, so client applications can display the real price of the transaction's fee. We assume the UCO price is given and aggregated by the network, so with consensus.

Exemple of payload
```js
{
  fee: 0.4942,
  uco_rates: {
    usd: 0.2,
    eur: 0.22
   }
 }
 ```

**Review by samuel-uniris — APPROVED** [review]

**Comment by redDwarf03** [comment]
hello
could you explain me why these fields are mandatory when I send a request to /api/transaction_fee ?
                  "originSignature"
                  "previousPublicKey" 
                  "previousSignature"

I would like to get the estimation of a transaction before sending to inform user how much could they pay to send tx.
so when I want to display estimation of fee, I have just the address of the owner, the balance, the amount and the recipient

thx

**Pull request merged**